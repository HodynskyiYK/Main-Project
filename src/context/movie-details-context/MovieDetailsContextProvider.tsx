import React, { createContext, FC, useContext, useState } from 'react'
import { IMovieDetailsContext, IMovieDetailsContextType } from './movieDetailsContextTypes'
import { IMovieItem } from '../../store/actions-types'

const MovieDetailsContext = createContext<IMovieDetailsContextType>({
    movieInfo: {} as IMovieItem,
    updateMovieDetails: (movie: IMovieItem) => {
        console.log(movie)
        throw new Error("Do not use context out of context provider")
    },
    movieDetailsState: false,
    hideMovieDetails: () => {
        throw new Error("Do not use context out of context provider")
    }
})

export const useMovieDetailsContext = () => useContext(MovieDetailsContext)

export const MovieDetailsContextProvider: FC<IMovieDetailsContext> = ({children}) => {
    const [movieDetailsState, setMovieDetailsState] = useState<boolean>(false)
    const [movieInfo, setMovieInfo] = useState<IMovieItem>({} as IMovieItem)

    const updateMovieDetails = (movie: IMovieItem) => {
        setMovieInfo({...movie})
        if (!movieDetailsState) {
            setMovieDetailsState(true)
        }
    }

    const hideMovieDetails = () => {
        if (movieDetailsState) {
            setMovieDetailsState(false)
            setMovieInfo({} as IMovieItem)
        }
    }

    return (
        <MovieDetailsContext.Provider value={{
            movieInfo,
            movieDetailsState,
            updateMovieDetails,
            hideMovieDetails
        }}>
            {children}
        </MovieDetailsContext.Provider>
    )
}
