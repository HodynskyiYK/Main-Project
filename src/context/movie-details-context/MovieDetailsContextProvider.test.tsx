import React from 'react'
import { render } from '@testing-library/react'
import { MovieDetailsContextProvider } from './MovieDetailsContextProvider'

describe('MovieDetailsContextProvider tests', () => {

    test('To match snapshot', () => {
        const {asFragment} = render(<MovieDetailsContextProvider>
            <div>Children</div>
        </MovieDetailsContextProvider>)
        expect(asFragment()).toMatchSnapshot()
    })
})