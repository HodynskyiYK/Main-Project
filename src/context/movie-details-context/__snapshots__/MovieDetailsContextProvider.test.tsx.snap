// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetailsContextProvider tests To match snapshot 1`] = `
<DocumentFragment>
  <div>
    Children
  </div>
</DocumentFragment>
`;
